orangecontrib/text/widgets/owcreatecorpus.py

```diff
@@ -0,0 +1,157 @@
+from Orange.widgets.widget import OWWidget, Output
+from AnyQt.QtWidgets import QGroupBox
+from PyQt5.QtCore import QSize, Qt
+from PyQt5.QtWidgets import (
+    QVBoxLayout,
+    QLineEdit,
+    QPlainTextEdit,
+    QSizePolicy,
+    QPushButton,
+)
+from orangewidget.settings import Setting
+
+from orangecontrib.text import Corpus
+
+
+class EditorsVerticalScrollArea(gui.VerticalScrollArea):
+    def __init__(self):
+        super().__init__(parent=None)
+        self.setSizePolicy(QSizePolicy.Expanding, QSizePolicy.Expanding)
+
+
+class CustomQPlainTextEdit(QPlainTextEdit):
+    editingFinished = Signal()
+
+    def focusOutEvent(self, _):
+        # TextEdit does not have editingFinished
+        self.editingFinished.emit()
+
+
+class DocumentEditor(QGroupBox):
+    text_changed = Signal(str, str)
+    remove_clicked = Signal()
+
+    def __init__(self, title, text, parent=None):
+        super().__init__(parent)
+
+        self.setLayout(QVBoxLayout())
+        self.title_le = QLineEdit()
+        self.title_le.setPlaceholderText("Document title")
+        self.title_le.setText(title)
+        self.title_le.editingFinished.connect(self._on_text_changed)
+        self.text_area = CustomQPlainTextEdit()
+        self.text_area.setPlaceholderText("Document text")
+        self.text_area.setPlainText(text)
+        self.text_area.editingFinished.connect(self._on_text_changed)
+
+        remove_button = QPushButton("x")
+        remove_button.setFixedWidth(35)
+        remove_button.setFocusPolicy(Qt.NoFocus)
+        remove_button.clicked.connect(self._on_remove_clicked)
+        box = gui.hBox(self)
+        box.layout().addWidget(self.title_le)
+        box.layout().addWidget(remove_button)
+        self.layout().addWidget(self.text_area)
+
+    def _on_text_changed(self):
+        self.text_changed.emit(self.title_le.text(), self.text_area.toPlainText())
+
+    def _on_remove_clicked(self):
+        self.remove_clicked.emit()
+
+
+class OWCreateCorpus(OWWidget):
+    name = "Create Corpus"
+    description = "Write/paste documents to create a corpus"
+    icon = "icons/CreateCorpus.svg"
+    priority = 120
+
+    class Outputs:
+        corpus = Output("Corpus", Corpus)
+
+    want_main_area = False
+
+    texts: List[Tuple[str, str]] = Setting([("", "")])
+    auto_commit: bool = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        self.editors = []
+
+        scroll_area = EditorsVerticalScrollArea()
+        self.editor_vbox = gui.vBox(self.controlArea, spacing=0)
+        self.editor_vbox.layout().setSpacing(10)
+        scroll_area.setWidget(self.editor_vbox)
+        self.controlArea.layout().addWidget(scroll_area)
+
+        for t in self.texts:
+            self._add_document_editor(*t)
+
+        add_btn = gui.button(
+            self.buttonsArea, self, "Add document", self._add_new_editor
+        )
+        add_btn.setFocusPolicy(Qt.NoFocus)
+        gui.auto_send(self.buttonsArea, self, "auto_commit")
```

Comment:
I think this needs to be changed to `auto_apply`, because if unchecked, `auto_send` will say "Send Selection", which is not the case here.

Reply:
Fixed